presto-tests/src/test/java/com/facebook/presto/execution/TestConcurrentQueriesInSingleTransaction.java

```diff
@@ -0,0 +1,86 @@
+ */
+package com.facebook.presto.execution;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.resourceGroups.ResourceGroupManagerPlugin;
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.facebook.presto.transaction.TransactionId;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.execution.QueryState.FAILED;
+import static com.facebook.presto.execution.QueryState.FINISHED;
+import static com.facebook.presto.execution.QueryState.RUNNING;
+import static com.facebook.presto.execution.TestQueryRunnerUtil.createQuery;
+import static com.facebook.presto.execution.TestQueryRunnerUtil.createQueryRunner;
+import static com.facebook.presto.execution.TestQueryRunnerUtil.waitForQueryState;
+import static com.facebook.presto.spi.ErrorType.USER_ERROR;
+import static com.facebook.presto.spi.StandardErrorCode.INVALID_CONCURRENT_QUERY;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static org.testng.Assert.assertEquals;
+
+public class TestConcurrentQueriesInSingleTransaction
+{
+    private static final String LONG_LASTING_QUERY = "SELECT COUNT(*) FROM lineitem";
+
+    @Test(timeOut = 60_000)
+    public void testConcurrentQueryInSingleTransaction()
+            throws Exception
+    {
+        try (DistributedQueryRunner queryRunner = createQueryRunner(ImmutableMap.of())) {
+            queryRunner.installPlugin(new ResourceGroupManagerPlugin());
+
+            Session clientSession = newTransactionSession();
+
+            // start transaction
+            QueryId startTxnQuery = createQuery(queryRunner, clientSession, "START TRANSACTION");
+
+            // wait until the transaction is registered
+            waitForQueryState(queryRunner, startTxnQuery, FINISHED);
+
+            assertEquals(queryRunner.getTransactionManager().getAllTransactionInfos().size(), 1);
+            TransactionId txnId = queryRunner.getTransactionManager().getAllTransactionInfos().get(0).getTransactionId();
+
+            // submit first "cli" query
+            QueryId firstCliQuery = createQuery(queryRunner, clientSession, txnId, LONG_LASTING_QUERY);
+
+            // wait for the first query to start
+            waitForQueryState(queryRunner, firstCliQuery, RUNNING);
+
+            // submit second "cli" query without waiting
+            QueryId secondCliQuery = createQuery(queryRunner, clientSession, txnId, LONG_LASTING_QUERY);
+
+            // the second query should failed
+            QueryInfo failedQueryInfo = queryRunner.getCoordinator().getQueryManager().getQueryInfo(secondCliQuery);
+            assertEquals(failedQueryInfo.getErrorType(), USER_ERROR);
+            assertEquals(failedQueryInfo.getErrorCode(), INVALID_CONCURRENT_QUERY.toErrorCode());
+
+            // Cancel the first query
+            TestQueryRunnerUtil.cancelQuery(queryRunner, firstCliQuery);
```

Comment:
If we do it automatically at shutdown, there will be several stack traces in the test log because of some sever exceptions. Adding these two lines simplifies the test log and makes it easier to debug when the test fails.